hw1/CBC-HMAC.py

```diff
@@ -51,15 +51,25 @@ def __strip_padding(self, data):
         :param data: input data
         :return: stripped data
         """
-        ?
+        pad_len = data[-1]
+        actual_data, padding = data[:-(pad_len + 1)], data[-(pad_len + 1): -1]
+        for byte in padding:
+            if byte != pad_len:
+                return None # invalid padding
+        expected_pad_len = self.block_len - ((len(actual_data) + 1) % self.block_len)
+        if expected_pad_len != pad_len:
+            return None # invalid padding
```

Comment:
Looking at the document now - I'm not sure that this has to be satisfied. The only condition is that the padded data's length is a multiple of the block size, but the padding itself can extend to multiple blocks (In the description of `padding_length` - "Legal values range from 0 to 255, inclusive.").

So I think we should check that `len(data)` is a multiple of `block_len`, and that the `padding_length` is between 0 and 255.

Reply:
Then also need to check that pad_len + len(actual_data) == len(data)